### PR TITLE
Handle `ordered` parameter in `CategoricalIndex.__repr__`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Thanks @mroeschke for catching this in https://github.com/rapidsai/cudf/pull/16665#discussion_r1735277661

This PR factors in the `ordered` parameter while generating the `repr` for `CategoricalIndex`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -1456,6 +1456,7 @@ def __repr__(self):
                 pd_preprocess.dtype._categories = (
                     preprocess.categories.to_pandas()
                 )
+                pd_preprocess.dtype._ordered = preprocess.dtype.ordered
```
Comment (mroeschke) [inline python/cudf/cudf/core/index.py:1459]: Does putting this in the `pd.Categorical` call trigger the validation that causes the slowdown too?

Comment (mroeschke) [reply python/cudf/cudf/core/index.py:1459]: Sorry, the `CategoricalIndex` call?

Comment (galipremsagar) [reply python/cudf/cudf/core/index.py:1459]: unfortunately assigning ordered in either or both of those calls don't result in `ordered` reflecting correctly in repr.

Comment (mroeschke) [reply python/cudf/cudf/core/index.py:1459]: Darn, thanks for checking.

I should have a patch that will help us avoid this in pandas 3.0 once https://github.com/pandas-dev/pandas/pull/59647 gets merged

**Review by mroeschke — APPROVED** [review]

**Pull request merged**